src/components/SearchBar/index.jsx

```diff
@@ -0,0 +1,98 @@
+import "./index.css";
+
+import React, { useContext, useEffect, useState } from "react";
+import { auth, db } from "../../lib/firebase";
+
+import { Box } from "@mui/material";
+import Post from "../Post";
+import { RowModeContext } from "../../hooks/useRowMode";
+import ShareModal from "../../reusableComponents";
+import SideBar from "../SideBar";
+
+function SearchBar() {
+  const [searchQuery, setSearchQuery] = useState("");
+  const [openShareModal, setOpenShareModal] = useState(false);
+  const [currentPostLink, setCurrentPostLink] = useState("");
+  const [postText, setPostText] = useState("");
+  const [posts, setPosts] = useState([]);
+  const rowMode = useContext(RowModeContext);
+
+  const handleSearch = (e) => {
+    setSearchQuery(e.target.value);
+    const filteredPosts = posts.filter((post) =>
+      post.post.username.toLowerCase().includes(searchQuery.toLowerCase())
+    );
+    setPosts(filteredPosts);
+  };
+
+  useEffect(() => {
+    const fetchPosts = async () => {
+      const postsRef = db.collection("posts");
```

Comment:
Also, this is an expensive call, use limits on the search and don't query the whole db.